ambry-api/src/main/java/com.github.ambry/clustermap/ClusterMap.java

```diff
@@ -51,6 +51,20 @@
   boolean hasDatacenter(String datacenterName);
 
   /**
+   * Gets the id for the given datacenter name, which is unique across the entire cluster.
+   * @param datacenterName The name of the datacenter to get the id.
+   * @return The id of the datacenter.
+   */
+  Short getDatacenterIdByName(String datacenterName);
```

Comment:
I only see it necessary to get the local datacenter's id (a one time call) in the workflow. Do you see a use for such a generic method?

Also, seems like a `byte` is sufficient for the id.